File: arc076_a.kt
import java.math.BigDecimal

fun main( args : Array<String> ) {
  val bbb   = 1000000007L
  val (a,b) = readLine()!!.split(" ").map { it.toInt() }
  val A     = listOf(1L, (1L..a).map { it.toLong() }. reduce {y,x -> 
                    (y%bbb*x%bbb)%bbb
                  } ).max()!!
  val B     = listOf(1L, (1L..b).map { it.toLong() }.reduce {y,x -> 
                    (y%bbb*x%bbb)%bbb
                  } ).max()!!
  when {
    a == b -> (A%bbb*B%bbb)%bbb*2%bbb
    a+1 == b || a == b+1 -> (A%bbb*B%bbb)%bbb
    else -> 0
  }.let { println(it) }
}

File: arc076_a.py
import os
from functools import reduce


a, b = map(int, input().split() )

aa = reduce(lambda y,x: y*x%1000000007, [i for i in range(1,a+1) ] )
bb = reduce(lambda y,x: y*x%1000000007, [i for i in range(1,b+1) ] )
if a == b :
  print(2*aa*bb%1000000007)
elif a+1 == b or a == b+1:
  print(aa*bb%1000000007)
else:
  print(0)

